### PR TITLE
Improving the bash script quality

### PR DESCRIPTION
**Pull request by vmx**
Some fixes, mostly based on a shellcheck run.

### CODE REVIEW

**Comment by vmx** [comment]
Anyone feel free to merge if it fits, I don't feel confident merging it as I don't know if there are any other things that should be merged first.

**Comment by vmx** [comment]
This was open for a long time and a lot of things changed. I did another run with shellcheck and improved a few things.

**Comment by laser** [comment]
@vmx - A wee small set o' requests:

1. Add shellcheck'ing to the `make lint` target
1. Modify CircleCI config to install shellcheck before running `make lint`

File: install-filecoin [hunk 1]
```diff
@@ -15,55 +16,47 @@ rust_sources_dir="rust"
...
-        tar -C "$__tmp_dir" -xzf "$__tarball_path"
+        tar -C "${__tmp_dir}" -xzf "${__tarball_path}"
 
         # copy build assets into root of filecoin-ffi
         #
         find -L "${__tmp_dir}" -type f -name filecoin.h -exec cp -- "{}" . \;
         find -L "${__tmp_dir}" -type f -name libfilecoin.a -exec cp -- "{}" . \;
         find -L "${__tmp_dir}" -type f -name filecoin.pc -exec cp -- "{}" . \;
 
+        check_installed_files
```
Comment (laser) [inline install-filecoin]: good catch

**Comment by vmx** [comment]
Thanks for merging :)

**Comment by laser** [comment]
but of course

**Pull request merged**